R/applications.R

```diff
@@ -237,3 +237,71 @@ showLogs <- function(appPath = getwd(), appFile = NULL, appName = NULL,
     cat(logs)
   }
 }
+
+#' Sync Application Metadata
+#'
+#' Update the metadata for requested application across all deployments
+#'
+#' @param appPath The path to the directory or file that was deployed.
+#'
+#' @note This function does not update metadata for Shiny and rpubs apps
+#'
+#' @export
+syncAppMetadata <- function(appPath) {
+  if (is.null(appPath) || !file.exists(appPath)) {
+    stop("appPath is null or does not exist")
+  }
+
+  # get all of the currently known deployments
+  deploys <- deployments(appPath)
+
+  syncWindow = getOption("rsconnect.metadata.sync.hours", 24) * 3600
+  now = as.numeric(Sys.time())
+
+  for (i in 1:nrow(deploys)) {
+    lastSyncTime <- deploys[i, "lastSyncTime"]
+
+    # for legacy dcf files that don't have sync time saved yet
+    if (is.null(lastSyncTime) || is.na(lastSyncTime))
+      lastSyncTime <- deploys[i, "when"]
+
+    # don't sync if within the configured time window
+    if (as.numeric(lastSyncTime) + syncWindow > now)
+      next
+
+    # don't sync non-connect apps
+    if (grepl("shinyapps.io", deploys[i, "hostUrl"], fixed = TRUE) ||
```

Comment:
We try to avoid hardcoding these hostnames as much as possible. The best way to skip apps is to pull the account info for the deployment and then check to see if we would use a Connect client to connect to it. You can see an example here:

https://github.com/rstudio/rsconnect/blob/79e297ecc600c1a00c32147601ba301e433ce312/R/applications.R#L46